src/bindings/python/flux/cli/base.py

```diff
@@ -1010,6 +1014,20 @@ def jobspec_create(self, args):
         """
         Create a jobspec from args and return it to caller
         """
+        plugins = {}
+        if args.plugin:
+            for provided_arg in args.plugin:
+                key, val = provided_arg.get_value()
+                if key in self.plugins.get_plugin_options():
+                    plugins[key] = val
+                else:
+                    raise ValueError(
+                        f"Unsupported option provided to -P/--plugin: {key}"
+                    )
```

Comment:
It might be better if this code was part of the CLIPluginRegistry class, which should take args and extract out the active plugins, raising an exception if there was any invalid input or a plugin raised an error in preinit.